libkbfs/bserver_remote.go

```diff
@@ -179,11 +124,121 @@ func (b *BlockServerRemote) ShouldRetry(rpcName string, err error) bool {
 }
 
 // ShouldRetryOnConnect implements the ConnectionHandler interface.
-func (b *BlockServerRemote) ShouldRetryOnConnect(err error) bool {
+func (b *blockServerRemoteClientHandler) ShouldRetryOnConnect(err error) bool {
 	_, inputCanceled := err.(libkb.InputCanceledError)
 	return !inputCanceled
 }
 
+var _ rpc.ConnectionHandler = (*blockServerRemoteClientHandler)(nil)
+
+// NewBlockServerRemote constructs a new BlockServerRemote for the
+// given address.
+func NewBlockServerRemote(config Config, blkSrvAddr string, ctx Context) *BlockServerRemote {
+	log := config.MakeLogger("BSR")
+	deferLog := log.CloneWithAddedDepth(1)
+	bs := &BlockServerRemote{
+		config:     config,
+		log:        log,
+		deferLog:   deferLog,
+		blkSrvAddr: blkSrvAddr,
+	}
+	bs.log.Debug("new instance server addr %s", blkSrvAddr)
+
+	// Use two separate auth tokens and clients -- one for writes and
+	// one for reads.  This allows small reads to avoid getting
+	// trapped behind large asynchronous writes.  TODO: use some real
+	// network QoS to achieve better prioritization within the actual
+	// network.
+	putClientHandler := &blockServerRemoteClientHandler{
+		bs:   bs,
+		name: "BlockServerRemotePut",
+	}
+	bs.putAuthToken = NewAuthToken(config,
```

Comment:
Just verifying -- separate TCP connections need separate auth tokens, i.e. the two connections can't just share one?